examples/pytorch/hetgnn/main.py

```diff
@@ -1,90 +1,672 @@
+                    for j in range(len(neigh_list_id)):
+                        v_p_list_train[node_id].append("p" + str(neigh_list_id[j]))
+
+    # store paper venue
+    p_v = [0] * config["P_n"]
+    with open(osp.join(config["data_dir"], "academic", "p_v.txt"), "r") as p_v_f:
+        for line in p_v_f:
+            line = line.strip()
+            p_id = int(re.split(",", line)[0])
+            v_id = int(re.split(",", line)[1])
+            p_v[p_id] = v_id
+
+    # paper neighbor: author + citation + venue
+    p_neigh_list_train = [[] for k in range(config["P_n"])]
+    for i in range(config["P_n"]):
+        p_neigh_list_train[i] += p_a_list_train[i]
+        p_neigh_list_train[i] += p_p_cite_list_train[i]
+        p_neigh_list_train[i].append("v" + str(p_v[i]))
+
+    return {
+        "a_p_list_train": a_p_list_train,
+        "a_p_list_test": a_p_list_test,
+        "p_a_list_train": p_a_list_train,
+        "p_a_list_test": p_a_list_test,
+        "p_p_cite_list_train": p_p_cite_list_train,
+        "p_p_cite_list_test": p_p_cite_list_test,
+        "p_neigh_list_train": p_neigh_list_train,
+        "v_p_list_train": v_p_list_train,
+        "p_v": p_v,
+    }
+
+
+def a_a_collaborate_train_test(config, input_data_map):
+    a_a_list_train = [[] for _ in range(config["A_n"])]
+    a_a_list_test = [[] for _ in range(config["A_n"])]
+    p_a_list = [input_data_map["p_a_list_train"], input_data_map["p_a_list_test"]]
+
+    for t in range(len(p_a_list)):
+        for i in range(len(p_a_list[t])):
+            for j in range(len(p_a_list[t][i])):
+                for k in range(j + 1, len(p_a_list[t][i])):
+                    if t == 0:
+                        a_a_list_train[int(p_a_list[t][i][j][1:])].append(
+                            int(p_a_list[t][i][k][1:])
+                        )
+                        a_a_list_train[int(p_a_list[t][i][k][1:])].append(
+                            int(p_a_list[t][i][j][1:])
+                        )
+                    else:
+                        if len(a_a_list_train[int(p_a_list[t][i][j][1:])]) and len(
+                            a_a_list_train[int(p_a_list[t][i][k][1:])]
+                        ):
+                            if (
+                                int(p_a_list[t][i][k][1:])
+                                not in a_a_list_train[int(p_a_list[t][i][j][1:])]
+                            ):
+                                a_a_list_test[int(p_a_list[t][i][j][1:])].append(
+                                    int(p_a_list[t][i][k][1:])
+                                )
+                            if (
+                                int(p_a_list[t][i][j][1:])
+                                not in a_a_list_train[int(p_a_list[t][i][k][1:])]
+                            ):
+                                a_a_list_test[int(p_a_list[t][i][k][1:])].append(
+                                    int(p_a_list[t][i][j][1:])
+                                )
+
```

Comment:
We need more documentation for what each variable contains since they are all 4 dimensional, also docs on what each file is for // rename file to make it clear. Also docstrings would help. Otherwise it is not very useful to have an example that no one can read.

Reply:
Also what is the need for this example that is not already covered by other examples? It seems not very useful to have such a complex example.

Reply:
This is outside of scope of this PR. I made this example self contained without changing training code.